manager/allocator/cnmallocator/drivers_ipam.go

```diff
@@ -1,14 +1,39 @@
 package cnmallocator
 
 import (
+	"net"
+	"strconv"
+	"strings"
+
 	"github.com/docker/libnetwork/drvregistry"
 	"github.com/docker/libnetwork/ipamapi"
 	builtinIpam "github.com/docker/libnetwork/ipams/builtin"
 	nullIpam "github.com/docker/libnetwork/ipams/null"
 	remoteIpam "github.com/docker/libnetwork/ipams/remote"
+	"github.com/docker/libnetwork/ipamutils"
+	"github.com/sirupsen/logrus"
 )
 
-func initIPAMDrivers(r *drvregistry.DrvRegistry) error {
+func initIPAMDrivers(r *drvregistry.DrvRegistry, defaultAddrPool []*net.IPNet, subnetSize int) error {
+	var addressPool []*ipamutils.NetworkToSplit
+	var str strings.Builder
+	str.WriteString("Subnetlist - ")
+	if defaultAddrPool != nil {
+		for _, p := range defaultAddrPool {
+			// trim leading and trailing white spaces
+			poolsDef := ipamutils.NetworkToSplit{}
+			poolsDef.Base = p.String()
+			poolsDef.Size = subnetSize
+			addressPool = append(addressPool, &poolsDef)
+			str.WriteString(p.String() + ",")
+		}
+		str.WriteString(": Size ")
+		str.WriteString(strconv.Itoa(subnetSize))
+	}
+	ipamutils.ConfigGlobalScopeDefaultNetworks(addressPool)
+	if addressPool != nil {
+		logrus.Infof("Swarm init global scope default address pool succeeded" + str.String())
```

Comment:
nit: This looks like it will produce output of the form "address pool succeededSubnetlist - ".  (i.e. no separator)   I'd probably also rephrase a little to something like "Swarm initialized global default address pool to: ".